targets/TARGET_TOSHIBA/TARGET_TMPM46B/trng_api.c

```diff
@@ -0,0 +1,105 @@
+/* mbed Microcontroller Library
+ * (C)Copyright TOSHIBA ELECTRONIC DEVICES & STORAGE CORPORATION 2017 All rights reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "mbed_error.h"
+#include "tmpm46B_esg.h"
+#include "trng_api.h"
+
+#define ESG_BUSY_SET                  (0x00000001U)
+
+volatile uint32_t CompleteFlag = 0U; // Global variables
+
+static void ESG_Config(void)
+{
+    uint16_t latchtiming;
+    uint16_t Fintming = 560U;
+    // Confirm the ESG core stops
+    if (ESG_GetCalculationStatus() == ESG_CALCULATION_COMPLETE) {
+        // Confirm no interrupt generation
+        if (ESG_GetIntStatus() == DISABLE) {
+            // Set the latch timing
+            ESG_SetLatchTiming(ESG_LATCH_TIMING_1);
+            // Get latchtiming value
+            latchtiming = ESG_GetLatchTiming();
+            if (Fintming >= 512U * (latchtiming + 1U) + 3U) {
+                // Set the output timing
+                ESG_SetFintiming(Fintming);
+            } else {
+                error("Fintming value input error !\n");
+            }
+        } else {
+            // Do nothing
+        }
+    } else {
+        // Do nothing 
+    }
+}
+
+void trng_init(trng_t *obj)
+{
+    (void)obj;
+
+    // Enable clock supply
+    CG_SetFcPeriphB(CG_FC_PERIPH_ESG,ENABLE);
+    // Enable interrupt
+    NVIC_ClearPendingIRQ(INTESG_IRQn);
+    NVIC_EnableIRQ(INTESG_IRQn);
+    // Initialise ESG core
+    ESG_Config();
+    // Start up the ESG core
+    ESG_Startup();
+}
+
+void trng_free(trng_t *obj)
+{
+    (void)obj;
+    // Disable interrupt
+    NVIC_DisableIRQ(INTESG_IRQn);
+    ESG_IPReset();
+    // Disable clock supply
+    CG_SetFcPeriphB(CG_FC_PERIPH_ESG,DISABLE);
+}
+
+int trng_get_bytes(trng_t *obj, uint8_t *output, size_t length, size_t *output_length)
+{
+    volatile uint8_t random[64] = {0};
+    uint8_t i;
+    *output_length = 0;
+    (void)obj;
+
+    while (TSB_ESG->ST == ESG_BUSY_SET) {
+        // wait for completing calculation
+    }
+    // check the interrupt flag 
+    if (CompleteFlag == 1) {
```

Comment:
How does this work for the first call to `trng_get_bytes`? We initialize `CompleteFlag = 0`, so unless the ESG fires an interrupt at startup (does it?), this path will never be taken.